docs/cn_omf.md

```diff
@@ -203,16 +214,92 @@ This is followed by the bloom filter.
 
 ### HyperLogLog
 
-This region does not have a separate header, just the hyperloglog (hlog) bytes.  The offset and length in the KBlock header points to this region.  Each kblock stores its hlog bytes and the kvset's hlog can be composed using the individual kblock hlogs.
+This region does not have a separate header, just the hyperloglog (hlog) bytes.
+The offset and length in the KBlock header points to this region. Each kblock
+stores its hlog bytes and the kvset's hlog can be composed using the individual
+kblock hlogs.
 
 ## VBlocks (Value blocks)
 
-VBlocks consist of a header followed by all the values.  The offset of a specifc value and its length are obtained from the kmd entry for this value which is linked to the corresponding key.
-
-### VBlock header
+VBlocks consist of a header followed by all the values. The offset of a specifc
+value and its length are obtained from the kmd entry for this value which is
+linked to the corresponding key.
+
+### VBlock trailer
+
+    +---------------------+
+    | u32 vbh_magic       |
+    | u32 vbh_version     |
+    | u64 vbh_vgroup      |
+    | u32 vbh_min_key_off |
+    | u32 vbh_min_key_len |
+    | u32 vbh_max_key_off |
+    | u32 vbh_max_key_len |
+    |---------------------|
+    | min key             | 1344 bytes
+    |---------------------|
+    | max key             | 1344 bytes
+    +---------------------+
+
+## Kvsets (Key-value sets)
+
+Kvsets are composed of kblocks and vblocks which are related to each other.
+
+### Kvset trailer
+
+Mblock consisting of kvset metadata.
+
+    +----------+--------+--------+-------+----------+
+    |          |        |        |       |          |
+    | Metadata | Kblock | Vblock | PTree | Hyperlog |
+    |  Header  |  Trim  | Index  |       |          |
+    |          |        | Adjust |       |          |
+    +----------+--------+--------+-------+----------+
+
+#### Metadata header
+
+    +---------------------------+
+    | u32 kmh_num_kblocks       |
+    | u32 kmh_num_vblocks       |
+    | u32 kmh_num_vgroups       |
+    | u32 kmh_kblk_trim_off     |
+    | u32 kmh_kblk_trim_len     |
+    | u32 kmh_vgroup_layout_off |
+    | u32 kmh_vgroup_layour_len |
+    | u32 kmh_vblk_idx_adj_off  |
+    | u32 kmh_vblk_idx_adj_len  |
+    | u32 kmh_ptree_off         |
+    | u32 kmh_ptree_len         |
+    | u32 kmh_hlog_off          |
+    | u32 kmh_hlog_len          |
+    +---------------------------+
+
+#### Kblock trim
+
+Kblock trim masks parts of a kblock's data. During a kvset split, a portion of
+a kblock will be hidden. Each segment is the trim associated with 1 kblock.
+
+    +------------------+
+    | ...              |
+    +------------------+
+    | u16 kt_start_pg  |
+    | u8  kt_start_off |
+    | u16 kt_end_pg    |
+    | u8  kt_end_off   |
+    +------------------+
+    | ...              |
+    +------------------+
```

Comment:
Do we need trim for every kblock?  Or start trim only for first kblock and end trim only for last kblock?

Reply:
Is u8 enough to index max number of keys in a WBT leaf node?

Reply:
In our discussion on Monday, you said 25, so I based it off that, but I don't know. I'll have to investigate unless someone knows off the top of their head.

Reply:
i think you need more bits.  leaf node is 4K.  wbt_node_hdr_omf is 12 bytes.  that leaves 4084 for N keys + N wbt_lfe_omf.  min key len is 1 byte, wbt_lfe_omf is 4 bytes.  so 4084/5 = max 816 keys in a leaf node.

Reply:
> Do we need trim for every kblock? Or start trim only for first kblock and end trim only for last kblock?

Good point. I think this might be the case.